### DISCUSSION TITLE
Some thoughts on the helm charts about deployment strategy and existingSecret name

### DISCUSSION THREAD

**Discussion by brujoand** [firefly-iii/firefly-iii#7518] [General]
Thanks for getting the helm charts up! It made the transition from the deprecated k8s-at-home charts much easier 👍 

I have some thoughts which could probably be have been Pull Requests instead, but I might not have all the details so I thought I'd. check.

The firefly-iii chart creates a deployment with a default strategy of rollingUpdate.  This is a problem because the default values create a persistent volume, and those tend to be ReadWriteOnce, so any update of the deployment will then hang until the old pod is manually deleted and releases the volume. I noticed that there is support for autoscaling so I assume firefly-iii works in HA? In any case it would probably be good to allow overriding the strategy from values.

The other thing is the 'existingSecret' functionality in the data-importer. I had setup the 1password operator to create secrets for me. I then named the secret `FIREFLY_III_ACCESS_TOKEN` which is what the application expects. But when using the existingSecret it assumes that the secret is called `accessToken`. So my `FIREFLY_III_ACCESS_TOKEN` was first [set correctly](https://github.com/firefly-iii/kubernetes/blob/45a5bfb98cefd90a24af3e1fd369b97180b10b8b/charts/importer/templates/deployment.yaml#LL59C11-L59C19) by the `envFrom:` section, but then [overwritten by the missing](https://github.com/firefly-iii/kubernetes/blob/45a5bfb98cefd90a24af3e1fd369b97180b10b8b/charts/importer/templates/deployment.yaml#L43) `accessToken` value. Maybe we should expect the secret to have the same name as the target variable `FIREFLY_III_ACCESS_TOKEN`?

**Comment by brujoand** [comment]
Actually, I just noticed that the there is another layer that confused me here about the existingSecret. When setting the accessToken it looks for `Values.fireflyiii.auth.existingSecret` not the secret used in `.Values.config.existingSecret`. So I guess for my case I would just like to disable the `.Values.fireflyiii.auth.existingSecret`.

**Comment by JC5** [comment]
Hey, thanks for these suggestions!

I'm not the one who developed or maintained (see the past tense?) kubernetes scripts. If you feel a PR is in order, please open one. Make sure you follow the instructions re: versioning etc.

**Reply by brujoand** [reply]
Cool, I'll keep playing with this then and see what improvements I can make 👍
